app/src/main/java/com/realityexpander/tasky/presentation/agenda_screen/AgendaState.kt

```diff
@@ -0,0 +1,32 @@
+package com.realityexpander.tasky.presentation.agenda_screen
+
+import com.realityexpander.tasky.presentation.util.UiText
+
+data class AgendaState(
+    val username: String = "",
+    val email: String = "",
+    val password: String = "",
+    val confirmPassword: String = "",
+    val isPasswordVisible: Boolean = false,
+
+    val isLoading: Boolean = false,
+    val isLoggedIn: Boolean = false,
+    val isError: Boolean = false,
+
+    val isInvalidUsername: Boolean = false,
+    val isShowInvalidUsernameMessage: Boolean = false,
+
+    val isInvalidEmail: Boolean = false,
+    val isShowInvalidEmailMessage: Boolean = false,
+
+    val isInvalidPassword: Boolean = false,
+    val isShowInvalidPasswordMessage: Boolean = false,
+
+    val isInvalidConfirmPassword: Boolean = false,
+    val isShowInvalidConfirmPasswordMessage: Boolean = false,
+
+    val isPasswordsMatch: Boolean = true,
+
+    val errorMessage: UiText = UiText.None,
+    val statusMessage: UiText = UiText.None,
```

Comment:
Also, here I'd rather work with nulls to indicate there is no error message. An empty error message still means an existing error to me.

Reply:
Good idea. I will convert these to nulls.